lib/Doctrine/DBAL/Connection.php

```diff
@@ -425,9 +425,46 @@ private function getDatabasePlatformVersion()
 
         // If not connected, we need to connect now to determine the platform version.
         if (null === $this->_conn) {
-            $this->connect();
+            try {
+                $this->connect();
+            } catch (\Exception $originalException) {
```

Comment:
@Ocramius I do not know what else we could catch here as the driver interface does not specify what to throw in case `$driver->connect()` fails. Any ideas welcome, though.